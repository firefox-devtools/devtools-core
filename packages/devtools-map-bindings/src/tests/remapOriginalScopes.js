/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

const { remapOriginalScopes } = require("../remapOriginalScopes");

describe("remapsOrignalScopes", () => {
  it("simple", () => {
    const scopes = [
      {
        type: "function",
        bindings: { n: [], e: [], o: [], u: [], r: [], t: [] }
      },
      {
        type: "script",
        bindings: {}
      }
    ];
    const originalScopes = [
      {
        type: "block",
        bindings: {
          na: { expr: "o", type: "var" },
          nb: { expr: "u", type: "var" }
        }
      },
      {
        type: "block",
        bindings: {
          ma: { expr: "r", type: "var" },
          mb: { expr: "t", type: "var" }
        }
      },
      {
        type: "function",
        displayName: "compare()",
        bindings: {
          a: { expr: "n", type: "arg" },
          b: { expr: "e", type: "arg" }
        }
      }
    ];

    expect(remapOriginalScopes(scopes, originalScopes)).toMatchSnapshot();
  });
});
