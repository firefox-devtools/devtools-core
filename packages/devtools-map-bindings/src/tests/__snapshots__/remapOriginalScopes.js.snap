// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remapsOrignalScopes simple 1`] = `
Object {
  "0": Object {
    "bindings": Object {
      "e": Array [],
      "n": Array [],
      "o": Array [],
      "r": Array [],
      "t": Array [],
      "u": Array [],
    },
    "type": "function",
  },
  "1": Object {
    "bindings": Object {},
    "type": "script",
  },
  "parent": null,
  "syntheticScopes": Object {
    "groupIndex": 0,
    "groupLength": 1,
    "scopes": Array [
      Object {
        "bindingsNames": Array [
          "na",
          "nb",
        ],
        "sourceBindings": Object {
          "na": "o",
          "nb": "u",
        },
        "type": "block",
      },
      Object {
        "bindingsNames": Array [
          "ma",
          "mb",
        ],
        "sourceBindings": Object {
          "ma": "r",
          "mb": "t",
        },
        "type": "block",
      },
      Object {
        "bindingsNames": Array [
          "a",
          "b",
        ],
        "sourceBindings": Object {
          "a": "n",
          "b": "e",
        },
        "type": "function",
      },
    ],
  },
}
`;
